Remove vendor/vendor-local dependencies

johngian opened the pull request:
* Remove vendor/vendor-local dependencies
* Use pypi requirements to fetch packages
* Migrate funfactory functionality to mozillians codebase
 * Move required ``base settings`` to mozillians settings
 * Move ``helpers/middleware/utils`` required by mozillians to our codebase

johngian commented:
TODO
 * Double-check ``./manage.py diffsettings`` differences between ``master`` and this branch
 * Double-check that deleted functionality from ``funfactory/manage.py`` is not affecting mozillians functionality
 * Fix `django_statsd.middleware.TastyPieRequestTimingMiddleware` incompatibility
 * Make sure that logging works after ``funfactory`` removal.

akatsoulas commented:
Have you considered using `deinit` to remove the submodules?

johngian commented:
@akatsoulas PR updated
 * Regarding submodules removal, I followed the steps from the docs. I think that it's OK now.
 * I changed our wsgi file to adapt to the changes
 * Added some missing settings entries
 * Squashed some commits with similar functionality

akatsoulas commented:
Please us `peep` instead of pip.

akatsoulas commented:
Also I would keep one file for the requirements and maybe one for dev purposes. I believe we can remove the `compiled.txt`.

akatsoulas commented on requirements/prod.txt:
```diff
@@ -1,9 +1,39 @@
+django==1.7.11
+django-compressor==1.4
+tower==0.4.1
+django-cronjobs==0.2.3
```
Is this used anywhere?

johngian replied:
```
> ./manage.py cron
Try one of these:

generate_humanstxt
index_all_profiles
invite
validate_fun_facts
```

akatsoulas replied:
:+1: 

Should we consider moving those in celery? Probably it's best not to do this in this PR. Maybe opening a new bug?

akatsoulas commented on mozillians/settings/base.py:
```diff
@@ -74,22 +138,71 @@
+    'django.core.context_processors.debug',
+    'django.core.context_processors.media',
+    'django.core.context_processors.request',
+    'session_csrf.context_processor',
```
Isn't this for `django-session-csrf`? Are we using it?

johngian replied:
https://github.com/mozilla/mozillians/blob/bca87d999c7067ef8f7d774302529d2e5bf0cef4/mozillians/common/monkeypatches.py#L36

akatsoulas replied:
Yes but `django-session-csrf` is not installed in this case (requirements file). Unless I have misunderstood something here

johngian replied:
requirements/prod.txt:django-session-csrf==0.5

akatsoulas commented on requirements/prod.txt:
```diff
@@ -1,9 +1,178 @@
+# sha256: mq4WtWKGak3apeiXhymrrbvtVEco2I4LnJr3sx3Qcu0
+django-nose==1.2
+# sha256: azCTiPTE-77saBPMs8MhujJF7yt94ETteyaFPsdcBfM
+sorl-thumbnail==11.12.1b0
```
I am getting an error from peep for this hash and this version

akatsoulas commented on requirements/dev.txt:
```diff
@@ -1,30 +1,140 @@
+simplegeneric==0.8.1
+# sha256: D_eMQD2bzPWkJabTGhKqa0fxwhyk3CVzp-LzKpczXrE
+# sha256: EF-NaGFvgkjiS_DpNy7wTTzBAQTxmA9U1Xss5zpa1Wo
+six==1.10.0
```
This is already declared in prod.txt. Please make sure that there aren't any duplicate packages in the two files.

akatsoulas commented:
One more commit please with the updates in the documentation :)

akatsoulas commented:
Regarding commit 6e61610, I don't have much to comment. Just make sure that all the helpers in there are used in the code base. Otherwise, please remove them.

akatsoulas commented on .gitmodules:
```diff
@@ -1,52 +1,3 @@
-	path = vendor-local/src/basket-client-submodule
-	url = git://github.com/mozilla/basket-client.git
 [submodule "mozillians/static/mozillians/css/less-prefixer"]
 	path = mozillians/static/mozillians/css/less-prefixer
```
I would prefer to completely remove every submodule from the project and provide all the necessary packages differently.

akatsoulas commented on .travis.yml:
```diff
@@ -11,12 +11,10 @@ before_install:
 install:
-  - pip install -r requirements/compiled.txt
-  - pip install -r requirements/dev.txt
+  - ./peep.sh install -r requirements/dev.txt
```
I would prefer all the changes to move in one of the two next commits. Let's make this commit all about removal. The reason I am proposing this is to make the changes easier to follow.

akatsoulas commented on mozillians/settings/base.py:
```diff
@@ -21,6 +38,13 @@
 
+# Repository directory
+ROOT = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+PROJECT_ROOT = os.path.dirname(os.path.dirname(__file__))
```
It seems that you are changing the root folder here.
Maybe you can try `os.path.abspath(os.path.join(__file__, '../../..'))

akatsoulas commented:
Reminder: We should update the chief script

akatsoulas commented on requirements/prod.txt:
```diff
@@ -1,9 +1,178 @@
+# sha256: gRBAtkfl1WhvhNtBXv1pfmJQAIsRK2kJunesBZ4UDHQ
+MySQL-python==1.2.5
+# sha256: YBKTHxpXJLUrDlNdRi-BxHwtFStdrl-kOEPTBtvPPcg
+kombu==2.1.8
```
Let's update `kombu` along with its dependencies since it's causing problems locally.

johngian commented:
@akatsoulas 
PR updated
* Removed ``less-prefixer`` submodule
* Updated kombu and dependencies
* Added deployment script (according to [1] and [2])
* Removed ``PROJECT_ROOT``

[1] https://github.com/mozilla/kuma/blob/master/scripts/chief_deploy.py
[2] https://github.com/mozilla/kitsune/blob/master/scripts/update/deploy.py

johngian commented:
Arrgg, forgot docs changes :(

johngian commented:
@akatsoulas PR updated with docs changes.

safwanrahman commented:
``peep`` is deprecated. The new version of ``pip`` has built in hash matching system. so better use that

johngian commented:
@safwanrahman Thanks for the tip. I suggest we keep peep for now, until these changes land to the servers. I don't want to add extra complexity on that and ``peep.sh`` seems to work on other projects.

We should definitely consider upgrading/porting hashes to pep ``1.8``.

:+1:

johngian commented:
@safwanrahman I cherry-picked some functionality from your PR since it was based on your initial work.

akatsoulas commented on mozillians/settings/base.py:
```diff
@@ -3,16 +3,33 @@
 # Django settings for the mozillians project.
 import logging
 import os.path
+import socket
```
Please add somewhere in this file the `SITE_URL` var.

johngian replied:
This wasn't neither in ``funfactory/settings_base.py`` nor in ``mozillians/settings/base.py``. Why add it now?

akatsoulas replied:
This is just a minor comment. The reason I am suggesting this, it's because is clearer, since this setting is needed, to have something like `SITE_URL = 'https://mozillians.org'` in the base settings file and overload it in the local.py depending on your needs. Feel free to ignore it though

johngian replied:
``SITE_URL`` is dependent to the deployment environment. I suggest we live it as it, and always define it in ``local.py``.

safwanrahman commented:
Seems like the requirements.txt also has the wheels. I think its better to have only the MD5 hashes.

akatsoulas commented:
I managed to run the site locally but `browserid` doesn't seem to work with the current configuration.

akatsoulas commented:
I did a quick test locally and everything looks good. Since this a major change, I would suggest to merge this PR to dev and fix any errors that may arise regarding deployment there. 

Please squash the last commit with the rest of the changes.
Really nice work r+wc :rocket: :beers:

akatsoulas commented:
oh and please make travis happy again

johngian commented:
Yeay travis is happy again! :rocket:
